src/runtime/entitlements-manager.js

```diff
@@ -853,6 +853,17 @@ export class EntitlementsManager {
         if (swgUserToken) {
           url = addQueryParam(url, 'sut', swgUserToken);
         }
+        // Add publisherProvidedId param for swg-basic
+        if (this.config_.publisherProvidedId) {
+          url = addQueryParam(url, 'ppid', this.config_.publisherProvidedId);
+        }
+        // Add publisherProvidedId param for swg-classic
+        else if (
+          params?.publisherProvidedId &&
+          typeof params.publisherProvidedId === 'string'
```

Comment:
Why does the type need to be checked here?

Reply:
+1 if we're going to do some type of assertion, let's do it above in the `getEntitlements` function and if we are going to ignore it we should give a `warn` on the page to let the developer know why

Reply:
The reason I was doing this check is because I was getting a JSC_TYPE_MISMATCH error when trying to using params.publisherProvidedId as a parameter in addQueryParam, saying it found : (string|undefined) and required: string. If this check is done in getEntitlements  to see if it is undefined I don't think a warn would get rid of the error? Would we want to throw an error?

Reply:
Spoke offline. We do a similar check below with `meteringStateId` which is the same thing as `publisherProvidedId` only masked in the metering configs. So this check is more for consistency with what we already do since its a param that the publisher can give to us directly